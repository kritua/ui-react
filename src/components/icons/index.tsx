import React, { SVGAttributes, ReactNode } from 'react';

export interface IProps extends SVGAttributes<SVGElement> {
    children?: ReactNode,
    className?: string
}

export const SVG = ({
    xmlns = 'http://www.w3.org/2000/svg',
    width = 24,
    height = 24,
    viewBox = '0 0 24 24',
    ...props
}: IProps) => (
    <svg
        xmlns={xmlns}
        width={width}
        height={height}
        viewBox={viewBox}
        {...props}
    />
);
