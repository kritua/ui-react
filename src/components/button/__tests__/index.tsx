import React from 'react';
import renderer from 'react-test-renderer';

import { Button } from './../index';
import { IconVideo } from './../../icons/video';
import { IconCircleMenu } from './../../icons/circle-menu';

const CHILDREN = 'Рыбный текст кнопки';

describe('Button', () => {
    it('Default', () => {
        const tree = renderer.create(<Button children={CHILDREN} />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('isLoader', () => {
        const tree = renderer.create(<Button children={CHILDREN} isLoading={true} />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('isCompact', () => {
        const tree = renderer.create(<Button children={CHILDREN} isCompact={true} />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('Icon + Text', () => {
        const tree = renderer.create(
            <Button>
                <IconVideo />
                {CHILDREN}
            </Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('Icon', () => {
        const tree = renderer.create(
            <Button>
                <IconVideo />
            </Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('Multi icon', () => {
        const tree = renderer.create(
            <Button>
                <IconVideo />
                <IconCircleMenu />
            </Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('SVG', () => {
        const tree = renderer.create(
            <Button>
                <svg xmlns="http://www.w3.org/2000/svg" width="18" height="18">
                    <path fill="#fff" d="m7.172 11.4 6.794-6.9L15 5.55 7.172 13.5 3 9.263l1.034-1.05L7.172 11.4Z" />
                </svg>
            </Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
