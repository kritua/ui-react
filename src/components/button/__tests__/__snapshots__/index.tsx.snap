// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Default 1`] = `
<button
  className="button button_medium button_default"
  type="button"
>
  Рыбный текст кнопки
</button>
`;

exports[`Button Icon + Text 1`] = `
<button
  className="button button_medium button_default"
  type="button"
>
  <svg
    height={24}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M3 5h12a2 2 0 0 1 2 2v1.382l6-3v13.236l-6-3V17a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V7a2 2 0 0 1 2-2Zm14 8.382 4 2V8.618l-4 2v2.764ZM3 7v10h12V7H3Z"
      fillRule="evenodd"
    />
  </svg>
  Рыбный текст кнопки
</button>
`;

exports[`Button Icon 1`] = `
<button
  className="button button_medium button_default button_compact"
  type="button"
>
  <svg
    height={24}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M3 5h12a2 2 0 0 1 2 2v1.382l6-3v13.236l-6-3V17a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V7a2 2 0 0 1 2-2Zm14 8.382 4 2V8.618l-4 2v2.764ZM3 7v10h12V7H3Z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`Button Multi icon 1`] = `
<button
  className="button button_medium button_default button_compact"
  type="button"
>
  <svg
    height={24}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M3 5h12a2 2 0 0 1 2 2v1.382l6-3v13.236l-6-3V17a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V7a2 2 0 0 1 2-2Zm14 8.382 4 2V8.618l-4 2v2.764ZM3 7v10h12V7H3Z"
      fillRule="evenodd"
    />
  </svg>
  <svg
    height={24}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M12 23C5.925 23 1 18.075 1 12S5.925 1 12 1s11 4.925 11 11-4.925 11-11 11Zm0-2a9 9 0 1 0 0-18 9 9 0 0 0 0 18Zm-5-5v-2h10v2H7Zm0-8v2h10V8H7Zm0 3v2h10v-2H7Z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`Button SVG 1`] = `
<button
  className="button button_medium button_default button_compact"
  type="button"
>
  <svg
    height="18"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m7.172 11.4 6.794-6.9L15 5.55 7.172 13.5 3 9.263l1.034-1.05L7.172 11.4Z"
      fill="#fff"
    />
  </svg>
</button>
`;

exports[`Button isCompact 1`] = `
<button
  className="button button_medium button_default button_compact"
  type="button"
>
  Рыбный текст кнопки
</button>
`;

exports[`Button isLoader 1`] = `
<button
  className="button button_medium button_default"
  disabled={true}
  tabIndex={-1}
  type="button"
>
  <svg
    className="loader loader_button"
    viewBox="0 0 50 50"
  >
    <circle
      className="loader__circle loader__circle_rich-grey"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      strokeWidth="3"
    />
  </svg>
</button>
`;
